jobs/mongodb_migration/src/mongodb_migration/collector.py

```diff
@@ -215,4 +215,11 @@ def get_migrations(self) -> List[Migration]:
                 version="20230524192300",
             ),
             MetricsDeletionMigration(job_type="/config-names", cache_kind="/config-names", version="20230524192400"),
+            MigrationQueueDeleteTTLIndex(
```

Comment:
Beware, this migration has never worked, by the way, because any old code that runs during the migration will recreate the old index, and we will have a conflict (same name, different parameters). I'm not sure what is the solution.

Reply:
I created a bug for it, maybe we can handle it in a separate PR https://github.com/huggingface/datasets-server/issues/1326

Reply:
I would like to have a try, my guess is that the issue is the usage of `mongoengine` with the desired schema is trying to re-create the index every time while doing validations https://github.com/huggingface/datasets-server/issues/1326#issuecomment-1581688644.
But since this is the last validation, I think no others will affect the index deletion.

Reply:
But the workers, the api service, the admin service (and even the backfill job, if running) are running during the migration, and they use the old version of queue.py, thus they recreate the index.
Also, I'm not sure if the index deletion is immediate (I think it takes some time once asked for).

Reply:
maybe we have to create a new index (have both in parallel)
then delete the old one in another migration, in a new deployment (so that no running app would try to recreate it)